src/test/java/com.teragrep.jos_01.procfs/ProcessTest.java

```diff
@@ -0,0 +1,192 @@
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs;
+
+import com.teragrep.jos_01.procfs.status.ProcessStat;
+import com.teragrep.jos_01.procfs.status.Statm;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class ProcessTest {
+
+    Process systemd;
+    Process kthreadd;
+
+    @Test
+    public void readSystemdFileTest() {
+        Process systemd = new Process(1);
+        Map<String, String> systemdStats = systemd.stat().statistics();
+        Assertions.assertEquals("1", systemdStats.get("pid"));
+        Assertions.assertEquals("(systemd)", systemdStats.get("comm"));
+        Assertions.assertEquals(52, systemdStats.size());
+        for (Map.Entry<String, String> entry : systemdStats.entrySet()) {
+            Assertions.assertNotNull(entry.getValue());
+            Assertions.assertNotNull(entry.getKey());
+        }
+    }
+
+    @Test
+    public void readKthreaddFileTest() {
+        Process kthreadd = new Process(2);
+        Map<String, String> kthreaddStats = kthreadd.stat().statistics();
+        Assertions.assertEquals("2", kthreaddStats.get("pid"));
+        Assertions.assertEquals("(kthreadd)", kthreaddStats.get("comm"));
+        Assertions.assertEquals(52, kthreaddStats.size());
+        for (Map.Entry<String, String> entry : kthreaddStats.entrySet()) {
+            Assertions.assertNotNull(entry.getValue());
+            Assertions.assertNotNull(entry.getKey());
+        }
+    }
+
+    @Test
+    public void constructorTest() {
+        Process process = new Process(1);
+        Process processString = new Process("1");
+        Assertions.assertNotNull(process.isAlive());
```

Comment:
Use `Assertions.assertTrue()` as that is more correct to use, that method should never return null

Reply:
Fixed